ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlSecurityExtension.java

```diff
@@ -34,21 +34,41 @@
  */
 public interface KsqlSecurityExtension extends AutoCloseable {
   /**
-   * Initializes the security extension. This is called in case the implementation requires
-   * to initializes internal objects prior to registering necessary endpoints.
-   *
-   * @param config The KSQL configuration containing security required configs.
-   * @throws KsqlException If an error occurs while initializing the security extension.
+   * @return The {@code KsqlAuthorizer} used to request access to KSQL resources.
    */
-  void initialize(KsqlConfig config) throws KsqlException;
+  KsqlAuthorizer getAuthorizer();
 
   /**
-   * Registers the security extension for the KSQL REST endpoints.
+   * Registers the security extension.
+   * </p>
+   * A {@link Configurable} is passed so that the extension can register REST filters to
+   * secure KSQL REST endpoints (i.e. Authorization filters).
    *
    * @param configurable The {@link Configurable} object where to register the security plugins.
+   * @param config The KSQL configuration containing security required configs.
    * @throws KsqlException If an error occurs while registering the REST security plugin.
    */
-  void registerRestEndpoints(Configurable<?> configurable) throws KsqlException;
+  void register(Configurable<?> configurable, KsqlConfig ksqlConfig);
+
+  /**
+   * Constructs a {@link org.apache.kafka.streams.KafkaClientSupplier} with the user's credentials.
+   *
+   * @param principal The {@link Principal} whose credentials will be used.
+   * @throws KsqlException If an error occurs while creating the
+   * {@link org.apache.kafka.streams.KafkaClientSupplier}.
+   */
+  KafkaClientSupplier getKafkaClientSupplier(Principal principal) throws KsqlException;
```

Comment:
These two methods shouldn't be here. (They're already in the file below.) Probably the result of a bad merge?

Reply:
Done